metadata-ingestion/src/datahub/sdk/lineage_client.py

```diff
@@ -220,16 +217,152 @@ def add_dataset_transform_lineage(
             cl.query = query_urn
             updater.add_fine_grained_upstream_lineage(cl)
 
-        # Throw if the dataset does not exist.
-        # We need to manually call .build() instead of reusing client.update()
-        # so that we make just one emit_mcps call.
+        # Throw if the dataset does not exist. TODO: is this valid?
```

Comment:
@hsheth2 should the lineage generation from sql fail when the table does not exist? or do we create one?